modules/dag-shared/src/main/scala/org/tessellation/dag/domain/block/DAGBlock.scala

```diff
@@ -1,41 +1,31 @@
 package org.tessellation.dag.domain.block
 
 import cats.data.NonEmptyList
-import cats.syntax.contravariant._
 import cats.syntax.reducible._
-import cats.syntax.semigroup._
-import cats.{Monoid, Order}
+
+import scala.collection.immutable.SortedSet
 
 import org.tessellation.ext.cats.syntax.next._
+import org.tessellation.ext.derevo.ordering
 import org.tessellation.schema.Fiber
 import org.tessellation.schema.height.Height
 import org.tessellation.schema.transaction.Transaction
 import org.tessellation.security.signature.Signed
 
-import derevo.cats.{eqv, show}
+import derevo.cats.{order, show}
 import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
 
 case class DAGBlockReference(parents: NonEmptyList[BlockReference])
 case class DAGBlockData(transactions: Set[Signed[Transaction]])
 
-@derive(encoder, decoder, eqv, show)
+@derive(encoder, decoder, order, ordering, show)
```

Comment:
FYI: I checked the usages of the existing `Order` instance, that I removed below and it was used only in a single place that required any deterministic order (not necessarily ordered by height as the first field), so I could safely replace it with generated `Order`.